http.go

```diff
@@ -152,3 +152,13 @@ func (ln tcpKeepAliveListener) Accept() (c net.Conn, err error) {
 	tc.SetKeepAlivePeriod(3 * time.Minute)
 	return tc, nil
 }
+
+func redirectToHTTPS(opts *Options, h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if opts.ForceHTTPS && r.TLS == nil {
```

Comment:
Out of curiosity, why not check `r.URL.Scheme`? Is `r.TLS` guaranteed to be filled at this point? I'm struggling to decipher the [docs](https://golang.org/pkg/net/http/#Request) for this field:

```
// TLS allows HTTP servers and other software to record
// information about the TLS connection on which the request
// was received. This field is not filled in by ReadRequest.
// The HTTP server in this package sets the field for
// TLS-enabled connections before invoking a handler;
// otherwise it leaves the field nil.
// This field is ignored by the HTTP client.
TLS *tls.ConnectionState
```

Reply:
@syscll `r.URL.Scheme` is actually just going to return an empty string. See this issue for details https://github.com/golang/go/issues/28940. `r.TLS` will not be nil if the TLS handshake has taken place, which happens at the very start of a https request, before the server phases of to the handler.

Reply:
I'm wondering what happens if the request is being proxied over http, this would end up in a redirect loop right? Do we argue at that point that the upstream proxy should be handling the redirect? 🤔 If that's what we decide we should document that